bin/csh/Makefile

```diff
@@ -121,8 +120,8 @@ build-tools: gethost
 DEPENDOBJS+= gethost
 gethost: gethost.c sh.err.h tc.const.h sh.h ${BUILD_TOOLS_META}
 	@rm -f ${.TARGET}
-	${BUILD_TOOLS_CC} ${BUILD_TOOLS_CFLAGS} ${BUILD_TOOLS_LDFLAGS} \
-	    ${TCSHDIR}/gethost.c -o ${.TARGET}
+	${CC:N${CCACHE_BIN}} -o gethost ${LDFLAGS} ${CFLAGS:C/-DHAVE_ICONV//} \
+	    ${TCSHDIR}/gethost.c
 .if make(build-tools)
```

Comment:
This hack here to `rm tc.const.h` is a local diff as well.  I've left it as it probably warrants being removed or upstreamed on its own (I think it is probably orthogonal to the BUILD_TOOLS changes, but maybe the FIXME below is relevant and so this should be removed as part of the BUILD_TOOLS_* commit?)